src/monitor/version_compat.h

```diff
@@ -14,38 +14,11 @@
 
 #include "postgres.h"
 
-/* we support Postgres versions 10, 11, 12, 13, 14, 15, and 16. */
-#if (PG_VERSION_NUM < 100000 || PG_VERSION_NUM >= 170000)
+/* we support Postgres versions 13, 14, 15, 16, and 17. */
+#if (PG_VERSION_NUM < 130000 || PG_VERSION_NUM >= 180000)
 #error "Unknown or unsupported postgresql version"
 #endif
 
-#if (PG_VERSION_NUM < 110000)
-
-#include "postmaster/bgworker.h"
-#include "utils/memutils.h"
-
-#define DEFAULT_XLOG_SEG_SIZE XLOG_SEG_SIZE
-
-#define BackgroundWorkerInitializeConnection(dbname, username, flags) \
-	BackgroundWorkerInitializeConnection(dbname, username)
-
-#define BackgroundWorkerInitializeConnectionByOid(dboid, useroid, flags) \
-	BackgroundWorkerInitializeConnectionByOid(dboid, useroid)
-
-#include "nodes/pg_list.h"
-
-typedef int (*list_qsort_comparator) (const void *a, const void *b);
-extern List * list_qsort(const List *list, list_qsort_comparator cmp);
-
-#endif
-
-#if (PG_VERSION_NUM < 120000)
-
-#define table_beginscan_catalog heap_beginscan_catalog
-#define TableScanDesc HeapScanDesc
-
-#endif
-
 #if (PG_VERSION_NUM >= 120000)
```

Comment:
I guess we can drop the #if, #endif around this since we are always >= 120000?

Reply:
While it's good to unclutter the CI and stop promising support for older Postgres versions (including in the docs and reference material), I see no reason to drop the actual code. If it doesn't incur development headaches, let's just keep that around.